bot/exts/evergreen/reddit.py

```diff
@@ -1,128 +1,366 @@
 
-            embed = discord.Embed()
-            page_tuple = (page_text, image_url)
-            pages.append(page_tuple)
+            return embed
+
+        if paginate:
+            return self.build_pagination_pages(posts, paginate=True)
+
+        # Use only starting summary page for #reddit channel posts.
+        embed.description = self.build_pagination_pages(posts, paginate=False)
+        embed.colour = Colour.blurple()
+        return embed
+
+    @loop()
+    async def auto_poster_loop(self) -> None:
+        """Post the top 5 posts daily, and the top 5 posts weekly."""
+        # once d.py get support for `time` parameter in loop decorator,
+        # this can be removed and the loop can use the `time=datetime.time.min` parameter
+        now = datetime.utcnow()
+        tomorrow = now + timedelta(days=1)
+        midnight_tomorrow = tomorrow.replace(hour=0, minute=0, second=0)
+
+        await sleep_until(midnight_tomorrow)
+
+        await self.bot.wait_until_guild_available()
+        if not self.webhook:
+            await self.bot.fetch_webhook(RedditConfig.webhook)
+
+        if datetime.utcnow().weekday() == 0:
+            await self.top_weekly_posts()
+            # if it's a monday send the top weekly posts
+
+        for subreddit in RedditConfig.subreddits:
+            top_posts = await self.get_top_posts(subreddit=subreddit, time="day")
+            username = sub_clyde(f"{subreddit} Top Daily Posts")
+            message = await self.webhook.send(username=username, embed=top_posts, wait=True)
+
+            if message.channel.is_news():
+                await message.publish()
+
+    async def top_weekly_posts(self) -> None:
+        """Post a summary of the top posts."""
+        for subreddit in RedditConfig.subreddits:
+            # Send and pin the new weekly posts.
+            top_posts = await self.get_top_posts(subreddit=subreddit, time="week")
+            username = sub_clyde(f"{subreddit} Top Weekly Posts")
+            message = await self.webhook.send(wait=True, username=username, embed=top_posts)
 
-            # ------------------------------------------------------------------
+            if subreddit.lower() == "r/python":
+                if not self.channel:
+                    log.warning("Failed to get #reddit channel to remove pins in the weekly loop.")
+                    return
+
+                # Remove the oldest pins so that only 12 remain at most.
+                pins = await self.channel.pins()
+
+                while len(pins) >= 12:
+                    await pins[-1].unpin()
+                    del pins[-1]
+
+                await message.pin()
+
+                if message.channel.is_news():
+                    await message.publish()
+
+    @group(name="reddit", invoke_without_command=True)
+    async def reddit_group(self, ctx: Context) -> None:
+        """View the top posts from various subreddits."""
+        await ctx.send_help(ctx.command)
```

Comment:
Because our help command isn't a `commands.HelpCommand`, this should use [`utils.extensions.invoke_help_command`](https://github.com/python-discord/sir-lancebot/blob/main/bot/utils/extensions.py#L36-L42).